homeassistant/components/mqtt/mixins.py

```diff
@@ -293,8 +293,9 @@ async def _async_discover_entities(event: Event | None) -> None:
             )
         )
 
-    hass.bus.async_listen("event_mqtt_reloaded", _async_discover_entities)
+    unsub = hass.bus.async_listen("event_mqtt_reloaded", _async_discover_entities)
     await _async_discover_entities(None)
+    return unsub
```

Comment:
unsub is not called at the platforms

Reply:
Platforms use `async_setup_platform_helper` which is a different method.